server/src/main/java/org/opensearch/search/wlm/tracker/QueryGroupUsageTracker.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.wlm.tracker;
+
+import org.opensearch.search.wlm.QueryGroupLevelResourceUsageView;
+
+import java.util.Map;
+
+/**
+ * This interface is mainly for tracking the queryGroup level resource usages
+ */
+public interface QueryGroupUsageTracker {
```

Comment:
I am unable to understand  a clear usage for this at this point. Ideally defining this kind of interface means you may have different kinds/implementations of "queryGroupUsage" tracker but I see you are implementing this in above service which looks confusing(and unnecessary). So how do we plan to this interface eventually?

Reply:
>you may have different kinds/implementations of "queryGroupUsage" tracker, it that the intention? 

Yes, i wanted to have this flexibility to include different implementations in the future.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/13897#discussion_r1700647054